jwst/cube_build/ifu_cube.py

```diff
@@ -1204,17 +1262,374 @@ def map_detector_to_outputframe(self, this_par1, this_par2,
                 coord1, coord2 = coord.radec2std(self.crval1,
                                                  self.crval2,
                                                  ra_use, dec_use)
+
                 if self.weighting == 'miripsf':
                     alpha_det = alpha[good_data]
                     beta_det = beta[good_data]
             elif self.coord_system == 'alpha-beta':
                 coord1 = alpha[good_data]
                 coord2 = beta[good_data]
 
-        return coord1, coord2, wave, flux, rois_det, roiw_det, weight_det, \
+        return coord1, coord2, wave, flux, slice, rois_det, roiw_det, weight_det, \
             softrad_det, alpha_det, beta_det
 # ********************************************************************************
 
+    def map_fov_to_dqplane(self, this_par1, coord1, coord2, wave, roiw_ave, slice):
+        """ Set an initial DQ flag for the IFU cube based on FOV of input data
+
+        Map the FOV of channel (MIRI) or slice (NIRSPEC) to the DQ plane
+        and set an initial DQ flagging. The process is different for MIRI and NIRSpec.
+        In the MIRI case all the slices map roughly to the same FOV across the
+        wavelength range covered by the IFU. This is not the case for NIRSpec the
+        30 different slices map to different FOV on the range of wavelengths.
+
+        Paramteter
+        ---------
+        this_par1: Channel (MIRI) or Grating (NIRSpec)
+        coord1: xi coordinates of input data (~x coordinate in IFU space)
+        coord2: eta coordinates of input data (~y coordinate in IFU space)
+        wave: wavelength of input data
+        roiw_ave: average spectral roi used to determine which wavelength bins
+            the input values would be mapped to
+        slice: integer slice value of input data (used in MIRI case to find
+            the points of the edge slices.)
+        """
+
+        # MIRI mapping:
+        # The FOV is roughtly the same for all the wavelenth ranges.
+        # The offset in the slices makes the calculation of the four corners
+        # of the FOV more comlicated. So we only use the two slices at
+        # the edges of the FOV to define the 4 corners.
+        # Note we can not use the wcs.footprint (because this footprint only
+        # consists of 4 values  ra min, ra max, dec min, dec max and we
+        # need 4 corners made of 8 different values.
+
+        if self.instrument == 'MIRI':
```

Comment:
Is there a guarantee somewhere earlier that the value of self.instrument will always be uppercase? If not, should add `.upper()`

Reply:
ok I added .upper() where it is read in  and setup - file_table.py